core/src/main/java/com/linecorp/armeria/internal/common/Http1ObjectEncoder.java

```diff
@@ -243,11 +243,16 @@ private ChannelFuture doWriteSplitData(int id, HttpData data, boolean endStream)
     }
 
     protected final ChannelFuture write(int id, HttpObject obj, boolean endStream) {
+        return write(id, obj, endStream, ch.newPromise());
```

Comment:
Note: we should be careful to call this method from non-header write methods only (with the exception of the `EMPTY_LAST_CONTENT` case)